sympy/core/mul.py

```diff
@@ -1264,26 +1266,38 @@ def _eval_is_odd(self):
         is_integer = self.is_integer
 
         if is_integer:
-            r, acc = True, 1
+            #The entire expression is an integer, but it
+            #can be decomposed into fractions (i.e.
+            #numerator and denominator.
+            #The counters keep track of multiples of 2 in
+            #numerator and denominator to determine oddness.
+            count_denom = 0
+            count_num = 0
             for t in self.args:
-                if not t.is_integer:
-                    return None
-                elif t.is_even:
-                    r = False
-                elif t.is_integer:
-                    if r is False:
-                        pass
-                    elif acc != 1 and (acc + t).is_odd:
-                        r = False
-                    elif t.is_odd is None:
-                        r = None
-                acc = t
-            return r
+                if t == S.Half:
+                    count_denom += 1
+                elif isinstance(t,Symbol):
```

Comment:
I thought the first if statement should handle that for the entire expression.  It will only enter into the other loop if is_integer  is True for the entire expression.  Then it moves on to evaluate the decomposition of the expression.  Does this logic work?